torch/distributed/_tensor/ops/other_ops.py

```diff
@@ -0,0 +1,149 @@
+    grad_input_stride.append(1)
+    dim_map = grad_output_spec.dim_map
+    sums = grad_output_spec.sums
+
+    grad_input_tensor_meta = TensorMeta(
+        torch.Size(input_sizes),
+        tuple(grad_input_stride),
+        grad_output_spec.tensor_meta.dtype,
+    )
+    grad_input_spec = DTensorSpec.from_dim_map(
+        grad_output_spec.mesh,
+        dim_map,
+        sums,
+        tensor_meta=grad_input_tensor_meta,
+    )
+
+    return OutputSharding(grad_input_spec)
+
+
+@register_prop_rule(aten.native_layer_norm.default)
+def layer_norm_rules(op_schema: OpSchema) -> OutputSharding:
+    input_spec, normalized_shape = op_schema.args_schema[0:2]
+
+    input_shape = input_spec.tensor_meta.shape
+    input_dim = len(input_shape)
+    norm_dim = len(normalized_shape)
+    mean_dim = input_dim - norm_dim
+    mean_shape = list(input_shape[0:mean_dim]) + [1 for _ in range(norm_dim)]
+    mean_stride = list(np.cumprod(mean_shape[::-1])[:-1][::-1])
+    mean_stride.append(1)
+    assert len(input_shape) > len(normalized_shape)
+    assert list(input_shape[-norm_dim:]) == normalized_shape
+
+    output_tensor_meta = input_spec.tensor_meta
+    mean_tensor_meta = TensorMeta(
+        torch.Size(mean_shape),
+        tuple(mean_stride),
+        input_spec.tensor_meta.dtype,
+    )
+    rstd_tensor_meta = TensorMeta(
+        torch.Size(mean_shape),
+        tuple(mean_stride),
+        input_spec.tensor_meta.dtype,
+    )
+
+    output_spec = DTensorSpec.from_dim_map(
+        input_spec.mesh,
+        [-1 for _ in range(input_dim)],
+        [],
+        tensor_meta=output_tensor_meta,
+    )
+
+    mean_spec = DTensorSpec.from_dim_map(
+        input_spec.mesh,
+        [-1 for _ in range(mean_dim)],
+        [],
+        tensor_meta=mean_tensor_meta,
+    )
+
+    rstd_spec = DTensorSpec.from_dim_map(
+        input_spec.mesh,
+        [-1 for _ in range(mean_dim)],
+        [],
+        tensor_meta=rstd_tensor_meta,
+    )
+
+    return OutputSharding([output_spec, mean_spec, rstd_spec])
+
+
+@register_prop_rule(aten.native_layer_norm_backward.default)
+def layer_norm_backward_rules(op_schema: OpSchema) -> OutputSharding:
+    input_spec = op_schema.args_schema[1]
+    weight_spec = op_schema.args_schema[5]
+    bias_spec = op_schema.args_schema[6]
+
+    grad_input_spec = DTensorSpec.from_dim_map(
+        input_spec.mesh,
+        [-1 for _ in range(len(input_spec.tensor_meta.shape))],
+        [],
+        tensor_meta=input_spec.tensor_meta,
+    )
+
+    grad_weight_spec = weight_spec
+    grad_bias_spec = bias_spec
+    return OutputSharding([grad_input_spec, grad_weight_spec, grad_bias_spec])
+
+
+@register_prop_rule(aten.bernoulli_.float)
+def bernoulli_rules(op_schema: OpSchema) -> OutputSharding:
+    input_spec = op_schema.args_schema[0]
+    return OutputSharding(input_spec)
+
+
+@register_prop_rule(aten.nll_loss_forward.default)
+def nll_loss_forward_rules(op_schema: OpSchema) -> OutputSharding:
```

Comment:
Given that those "other ops" are not related to convolution specifically, it would be better if we can separate the other ops enablement with convolution itself with some tests

Reply:
Can you elaborate more on this? Currently we need to register `slice_backward`, `bernoulli_`, `nll_loss_forward` and `nll_loss_backward` within [other_ops.py](https://github.com/KingsleyLiu-NV/pytorch/blob/fea_dtensor_conv/torch/distributed/_tensor/ops/other_ops.py) to run the tp training example [convnext_example.py](https://github.com/KingsleyLiu-NV/pytorch/blob/fea_dtensor_conv/torch/distributed/_tensor/examples/convnext_example.py).

Is it good to keep other_ops.py as it is, and add some unit tests by creating a new file `test_other_ops.py` within [test/distributed/_tensor](https://github.com/KingsleyLiu-NV/pytorch/tree/fea_dtensor_conv/test/distributed/_tensor)?

Reply:
I have add unit tests for other ops and showcase the reference log in the description in this PR.